python/tests/hypothesis/arcticdb/test_sort_merge.py

```diff
@@ -1,71 +1,235 @@
+
+def assert_cannot_finalize_without_staged_data(lib, symbol, mode):
+    with pytest.raises(UserInputException) as exception_info:
+        lib.sort_and_finalize_staged_data(symbol, mode=mode)
+    assert "E_NO_STAGED_SEGMENTS" in str(exception_info.value)
+  
+def assert_nat_is_not_supported(lib, symbol, mode):
+    with pytest.raises(UnsortedDataException) as exception_info:
+        lib.sort_and_finalize_staged_data(symbol, mode=mode)
+    assert "E_UNSORTED_DATA" in str(exception_info.value)
+
+def assert_staged_columns_are_compatible(lib, symbol, mode):
+    with pytest.raises(SchemaException) as exception_info:
+        lib.sort_and_finalize_staged_data(symbol, mode)
+    assert "E_DESCRIPTOR_MISMATCH" in str(exception_info.value)
+
+def has_nat_in_index(segment_list):
+    return any(pd.NaT in segment.index for segment in segment_list)
+
+def merge_and_sort_segment_list(segment_list, int_columns_in_df=None):
+    merged = pd.concat(segment_list)
+    # pd.concat promotes dtypes. If there are missing values in an int typed column
+    # it will become float column and the missing values will be NaN.
+    int_columns_in_df = int_columns_in_df if int_columns_in_df else []
+    for col in int_columns_in_df:
+        merged[col] = merged[col].replace(np.NaN, 0)
+    merged.sort_index(inplace=True)
+    return merged
+
+def assert_appended_data_does_not_overlap_with_storage(lib, symbol):
+    with pytest.raises(UnsortedDataException) as exception_info:
+        lib.sort_and_finalize_staged_data(symbol, mode=StagedDataFinalizeMethod.APPEND)
+
+def segments_have_compatible_schema(segment_list):
+    """
+    Used to check dynamic schemas. Considers all numeric types for compatible.
+    """
+    dtypes = {}
+    for segment in segment_list:
+        for col in segment:
+            if col not in dtypes:
+                dtypes[col] = segment.dtypes[col]
+            elif not are_dtypes_compatible(dtypes[col], segment.dtypes[col]):
+                return False
+    return True
+
+def get_append_keys(lib, sym):
+    lib_tool = lib._nvs.library_tool()
+    keys = lib_tool.find_keys_for_symbol(KeyType.APPEND_DATA, sym)
+    return keys
+
+@use_of_function_scoped_fixtures_in_hypothesis_checked
+@settings(deadline=None)
+@given(df_list=generate_dataframes(COLUMN_DESCRIPTIONS))
+def test_sort_merge_static_schema_write(lmdb_library, df_list):
+    lib = lmdb_library
+    sym = "test_sort_merge_static_schema_write"
+    for df in df_list:
+        lib.write(sym, df, staged=True, validate_index=False)
+    if len(df_list) == 0:
+        assert_cannot_finalize_without_staged_data(lib, sym, StagedDataFinalizeMethod.WRITE)
+        assert len(get_append_keys(lib, sym)) == 0
+        return
+    if not all(df_list[0].dtypes.equals(segment.dtypes) for segment in df_list):
+        assert_staged_columns_are_compatible(lib, sym, StagedDataFinalizeMethod.WRITE)
+        assert len(get_append_keys(lib, sym)) == 0
```

Comment:
Could/should this assert (and the similar ones below) be rolled into the `assert_staged_columns_are_compatible`/`assert_nat_is_not_supported`/`assert_appended_data_does_not_overlap_with_storage` functions?

Reply:
Yep, makes sense.